samples/memfs/memfs.go

```diff
@@ -627,7 +627,7 @@ func (fs *memFS) OpenFile(
 		// Set attribute (name=fileOpenFlagsXattr, value=OpenFlags) to test whether
 		// we set OpenFlags correctly. The value is checked in test with getXattr.
 		value := make([]byte, 4)
-		binary.LittleEndian.PutUint32(value, uint32(op.OpenFlags))
+		binary.LittleEndian.PutUint32(value, uint32(op.OpenFlags)&syscall.O_ACCMODE)
```

Comment:
The tests for the open flags were broken for me because the 0x8000 bit seemed to be implicitly set somewhere (this test failed against master branch before my other changes were applied). I think this corresponds to the O_EVTONLY flag, which is maybe related to how golang opens files for reading with EPOLL?

Reply:
I can reproduce this here. I’m not sure either where the extra 0x8000 bit comes from :-/

It would be interesting to git bisect the Linux kernel and track down when this behavior changed.